WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -688,25 +727,46 @@ import Foundation
         }
     }
 
-    func backfillItems(success:((hasMore: Bool) -> Void)?, failure: ((error: NSError) -> Void)?) {
+    func syncItemsForGap(success:((hasMore: Bool) -> Void)?, failure: ((error: NSError) -> Void)?) {
+        assert(syncIsFillingGap)
+        assert(indexPathForGapMarker != nil)
+
+        let post = tableViewHandler.resultsController.objectAtIndexPath(indexPathForGapMarker!) as? ReaderGapMarker
+        if post == nil {
+            // failsafe
+            return
+        }
+
+        // Reload the gap cell so it will start animating.
+        tableView.reloadRowsAtIndexPaths([indexPathForGapMarker!], withRowAnimation: .None)
+
         let syncContext = ContextManager.sharedInstance().newDerivedContext()
         let service =  ReaderPostService(managedObjectContext: syncContext)
+        let sortDate = post!.sortDate
 
         syncContext.performBlock {[weak self] () -> Void in
-
-            do  {
+            do {
                 let topic = try syncContext.existingObjectWithID(self!.readerTopic!.objectID) as! ReaderAbstractTopic
-
-                service.backfillPostsForTopic(topic,
-                    success: { (count:Int, hasMore:Bool) -> Void in
+                service.fetchPostsForTopic(topic,
+                    earlierThan:sortDate,
+                    deletingEarlier:true,
+                    success: {[weak self] (count:Int, hasMore:Bool) in
                         dispatch_async(dispatch_get_main_queue(), {
```

Comment:
(No need to change anything!). `dispatch_async` can receive the (last) block this way:

```
dispatch_async(dispatch_get_main_queue()) {
   // Your code
}
```